presto-main/src/main/java/com/facebook/presto/execution/ExecutionFailureInfo.java

```diff
@@ -52,7 +54,8 @@ public ExecutionFailureInfo(
             @JsonProperty("suppressed") List<ExecutionFailureInfo> suppressed,
             @JsonProperty("stack") List<String> stack,
             @JsonProperty("errorLocation") @Nullable ErrorLocation errorLocation,
-            @JsonProperty("errorCode") @Nullable ErrorCode errorCode)
+            @JsonProperty("errorCode") @Nullable ErrorCode errorCode,
+            @JsonProperty("destinationHost") @Nullable HostAddress destinationHost)
```

Comment:
Execution can fail for several reasons, only one of which has a 'destination host'.  This needs to be re-thought.